cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala

```diff
@@ -350,17 +679,47 @@ class QueryPlanTest extends DocumentingTestBase with SoftReset {
     profileQuery(
       title = "Limit",
       text =
-        """Returns the first 'n' rows from the incoming input.""".stripMargin,
+        """The `Limit` operator returns the first 'n' rows from the incoming input.""".stripMargin,
       queryText = """MATCH (p:Person) RETURN p LIMIT 3""",
       assertions = (p) => assertThat(p.executionPlanDescription().toString, containsString("Limit"))
     )
   }
 
+  @Test def lock() {
+    profileQuery(
+      title = "Lock",
+      text =
+        """The `Lock` operator locks the start and end node when creating a relationship.""".stripMargin,
+      queryText = """MATCH (s:Person {name: 'me'}) MERGE (s)-[:FRIENDS_WITH]->(s)""",
+      assertions = (p) => assertThat(p.executionPlanDescription().toString, containsString("Lock"))
+    )
+  }
+
+  @Test def optional() {
+    profileQuery(
+      title = "Optional",
+      text =
+        """xxx For use in optional match.""".stripMargin,
```

Comment:
The `Optional` operator is used to solve some `OPTIONAL` queries. It will pull data from it's source, and simply pass it through if any data exists. If nothing is returned by it's source, it will yield a single row with all columns set to `null`.

Reply:
👍